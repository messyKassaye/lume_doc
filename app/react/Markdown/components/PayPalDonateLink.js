import { Link } from 'react-router';
import PropTypes from 'prop-types';
import React from 'react';

const PayPalDonateLink = ({ paypalid, classname, children, currency, amount }) => {
  const amountParam = amount ? `&amount=${amount}` : '';
  const url = `https://www.paypal.com/cgi-bin/webscr?cmd=_donations&business=${paypalid}&currency_code=${currency}${amountParam}&source=url`;
  classname += ' paypal-donate';
  return (
    <Link className={classname} href={url} target="_blank" rel="noreferrer noopener">
      {children}
    </Link>
  );
};

PayPalDonateLink.defaultProps = {
  children: '',
  classname: '',
  amount: null,
};

PayPalDonateLink.propTypes = {
  paypalid: PropTypes.string.isRequired,
  currency: PropTypes.string.isRequired,
  classname: PropTypes.string,
  amount: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
    PropTypes.string,
  ]),
};

export default PayPalDonateLink;
