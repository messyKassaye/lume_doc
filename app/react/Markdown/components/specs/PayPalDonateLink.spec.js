import React from 'react';
import { shallow } from 'enzyme';
import PayPalDonateLink from '../PayPalDonateLink.js';

describe('Link', () => {
  beforeEach(() => {
    spyOn(console, 'warn');
  });
  it('should render a react-router Link with target _blank', () => {
    const component = shallow(
      <PayPalDonateLink currency="EUR" paypalid="1234">
        label
      </PayPalDonateLink>
    );
    expect(component).toMatchSnapshot();
  });

  it('should allow to set a custom amount', () => {
    const component = shallow(
      <PayPalDonateLink currency="EUR" paypalid="1234" amount="10.15">
        label
      </PayPalDonateLink>
    );
    expect(component).toMatchSnapshot();
  });

  it('should allow to set a custom class to the component', () => {
    const component = shallow(
      <PayPalDonateLink currency="EUR" paypalid="1234" className="custom class">
        label
      </PayPalDonateLink>
    );
    expect(component).toMatchSnapshot();
  });

  it('should allow nesting children inside', () => {
    const component = shallow(
      <PayPalDonateLink currency="EUR" paypalid="1234">
        <div>Extra content</div>
        <span>Label</span>
      </PayPalDonateLink>
    );
    expect(component).toMatchSnapshot();
  });
});
