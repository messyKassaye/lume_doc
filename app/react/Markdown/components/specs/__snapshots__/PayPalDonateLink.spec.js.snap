// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should allow nesting children inside 1`] = `
<Link
  className=" paypal-donate"
  href="https://www.paypal.com/cgi-bin/webscr?cmd=_donations&business=1234&currency_code=EUR&source=url"
  onlyActiveOnIndex={false}
  rel="noreferrer noopener"
  style={Object {}}
  target="_blank"
>
  <div>
    Extra content
  </div>
  <span>
    Label
  </span>
</Link>
`;

exports[`Link should allow to set a custom amount 1`] = `
<Link
  className=" paypal-donate"
  href="https://www.paypal.com/cgi-bin/webscr?cmd=_donations&business=1234&currency_code=EUR&amount=10.15&source=url"
  onlyActiveOnIndex={false}
  rel="noreferrer noopener"
  style={Object {}}
  target="_blank"
>
  label
</Link>
`;

exports[`Link should allow to set a custom class to the component 1`] = `
<Link
  className=" paypal-donate"
  href="https://www.paypal.com/cgi-bin/webscr?cmd=_donations&business=1234&currency_code=EUR&source=url"
  onlyActiveOnIndex={false}
  rel="noreferrer noopener"
  style={Object {}}
  target="_blank"
>
  label
</Link>
`;

exports[`Link should render a react-router Link with target _blank 1`] = `
<Link
  className=" paypal-donate"
  href="https://www.paypal.com/cgi-bin/webscr?cmd=_donations&business=1234&currency_code=EUR&source=url"
  onlyActiveOnIndex={false}
  rel="noreferrer noopener"
  style={Object {}}
  target="_blank"
>
  label
</Link>
`;
